File: Atividade-migracao-pratica/atividade2.py
class Atividade:
  def __init__(self, descricao):
    self.descricao = descricao
    self.concluida = False

  def marcar_como_concluida(self):
    self.concluida = True;

  def status(self):
    return '[X]' if self.concluida else '[ ]'

def sistema_de_atividades():
  lista = []
  while True:
    print('\n=== SISTEMA DE GESTÃO DE ATIVIDADES ===\n')
    print('1 - Adicionar atividade')
    print('2 - Listar atividades')
    print('3 - Concluir atividade')
    print('0 - Sair\n')

    opcao = input('Escolha uma opçao: ')

    if opcao == '1':
      descricao = input('Digite a descrição da atividade: ')
      nova_atividade = Atividade(descricao)
      lista.append(nova_atividade)
      print('Atividade adicionada!')

    elif opcao == '2':
      print('\n--- Lista de atividades ---')
      for index, atividade in enumerate(lista):
        print(f'{index + 1}. {atividade.status()} {atividade.descricao}')

    elif opcao == '3':
      print('\n--- Lista de atividades ---')
      for index, atividade in enumerate(lista):
        print(f'{index + 1}. {atividade.status()} {atividade.descricao}')
      escolher_atividade = int(input('\nDigite o número da atividade a concluir: '))
      if escolher_atividade >= 1 and escolher_atividade <= len(lista):
        lista[escolher_atividade - 1].marcar_como_concluida()
        print('Atividade concluida!')
      else:
        print('Número inválido.')

    elif opcao == '0':
      print('Encerrando sistema...')
      break
    else:
      print('Opção inválida!')

sistema_de_atividades()

File: Atividade-migracao-pratica/atividade2.java
import java.util.*;

class Atividade {
    String descricao;
    boolean concluida;

    Atividade(String descricao) {
        this.descricao = descricao;
        this.concluida = false;
    }

    void marcarComoConcluida() {
        this.concluida = true;
    }

    @Override
    public String toString() {
        return (concluida ? "[X] " : "[ ] ") + descricao;
    }
}

class SistemaAtividades {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<Atividade> lista = new ArrayList<>();
        int opcao;

        do {
            System.out.println("\n=== SISTEMA DE GESTÃO DE ATIVIDADES ===");
            System.out.println("1 - Adicionar atividade");
            System.out.println("2 - Listar atividades");
            System.out.println("3 - Concluir atividade");
            System.out.println("0 - Sair");
            System.out.print("Escolha uma opção: ");
            opcao = scanner.nextInt();
            scanner.nextLine(); // limpar o buffer

            switch (opcao) {
                case 1:
                    System.out.print("Digite a descrição da atividade: ");
                    String desc = scanner.nextLine();
                    lista.add(new Atividade(desc));
                    System.out.println("Atividade adicionada.");
                    break;
                case 2:
                    System.out.println("\n--- Lista de Atividades ---");
                    for (int i = 0; i < lista.size(); i++) {
                        System.out.println((i + 1) + " - " + lista.get(i));
                    }
                    break;
                case 3:
                    System.out.print("Digite o número da atividade a concluir: ");
                    int num = scanner.nextInt();
                    if (num >= 1 && num <= lista.size()) {
                        lista.get(num - 1).marcarComoConcluida();
                        System.out.println("Atividade concluída!");
                    } else {
                        System.out.println("Número inválido.");
                    }
                    break;
                case 0:
                    System.out.println("Encerrando sistema...");
                    break;
                default:
                    System.out.println("Opção inválida.");
            }
        } while (opcao != 0);

        scanner.close();
        }
}
